Lib/test/test_pstats.py

```diff
@@ -30,12 +34,39 @@ class StatsTestCase(unittest.TestCase):
     def setUp(self):
         stats_file = support.findfile('pstats.pck')
         self.stats = pstats.Stats(stats_file)
+        to_compile = 'import os'
+        self.temp_storage = tempfile.NamedTemporaryFile(delete=False)
+        profiled = compile(to_compile, '<string>', 'exec')
+        cProfile.run(profiled, filename=self.temp_storage.name)
+
+    def tearDown(self):
+        self.temp_storage.close()
+        os.remove(self.temp_storage.name)
 
     def test_add(self):
         stream = StringIO()
         stats = pstats.Stats(stream=stream)
         stats.add(self.stats, self.stats)
 
+    def test_dump_and_load_works_correctly(self):
+        temp_storage_new = tempfile.NamedTemporaryFile(delete=False)
```

Comment:
a new file is created and copied from the file created in `setUp`

Reply:
Protect your file creation with `try ... finally ...` or the file will be abandoned there if the assertion fails.

Reply:
sure